### ISSUE TITLE
Signature Verification Problem in script_tests.json

### ISSUE THREAD

**Issue by ilozanof** [bitcoin/bitcoin#13593]
Hi, I'm getting a FALSE as a result in ALL the Signature Verification test cases in "script_tests.json". Some tests related to the CHECKSIG or CHECKSMULTISIG pass because they are negative tests, but in all cases where some real signature is checked, the result is FALSE. Full disclosure: I'm running a different project in Java, and we are trying to implement all the test uses cases as they are described here in bitcoin-core.
I don't think that the test cases are wrong themselves, but on the other hand the crypto lib from spongycastle doesn't lie...
I'm sure I must be missing something, in case you ran across the same issue, any help is really appreciated.

Thanks

**Comment by MarcoFalke** [comment]
Please provide exact steps to reproduce.

**Comment by ilozanof** [comment]
All the signature verification failed.  I don't think this issue is reproducible here since I'm working with another tech stack (Java), so the crypto libraries might be different. Since the Private and Public Key are hardcoded in the test, the Transaction HASH that is generated during execution time seemed to be the only reasonable point of failure. 
I've created another set of test cases on my own, using different private/Pub key pairs, and the new tests are working fine, verifying the signature properly. So my guess is either the transaction serialization algorithm has been changed at some point and not reflected in my code, or this serialisation works in a different way in another tech environment (I don't think this is the case though).

Now it's fixed in my env, I didn't close the issue myself before because I'm still in the process of replicating all the scenarios.

Thx
